multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/v2/ApplicationRoutesCloudModelBuilder.java

```diff
@@ -53,12 +54,25 @@ private Set<CloudRoute> addExistingRoutes(Set<CloudRoute> routes, DeployedMtaApp
         if (deployedMtaApplication == null) {
             return routes;
         }
+        List<CloudRoute> existingRoutes = client.getApplicationRoutes(deployedMtaApplication.getGuid());
+        Set<CloudRoute> mergedRoutes = new HashSet<>();
+        for (CloudRoute route : routes) {
+            if (route.getRequestedProtocol() == null) {
+                mergedRoutes.add(getExistingRouteOrReturnNew(route, existingRoutes));
+            } else {
+                mergedRoutes.add(route);
+            }
```

Comment:
This is required because of the keep-existing-routes feature. If you specify it and the route already exists in the descriptor (without protocol, but the protocol of the route and app is http2) and perform a bg-deploy, it will wipe out the protocol